src/store/modules/courses.js

```diff
@@ -26,49 +26,60 @@ const courses = {
     },
     getCourseById(state, payload) {
       state.courseById = payload;
+    },
+    errorMessage(state, payload) {
+      state.error = payload;
     }
   },
   actions: {
     async showCourses(context) {
-      const response = await fetch('http://localhost:7000/api/courses');
-      const result = await response.json();
-      context.commit('updateCourses', result);
+      try {
+        const response = await axios.get('http://localhost:7000/api/courses');
+        const result = response.data;
+        context.commit('updateCourses', result);
+      } catch (error) {
+        context.commit('errorMessage', 'Unable to get courses right now try after some time');
+      }
     },
     async addCourse(context, payload) {
```

Comment:
remove context and take only those objects that you require like { commit }

Reply:
done